packages/wrangler/src/cloudchamber/common.ts

```diff
@@ -177,7 +177,8 @@ export async function promiseSpinner<T>(
 		message: "Loading",
 	}
 ): Promise<T> {
-	if (json) {
+	const interactive = !json && !isNonInteractiveOrCI();
+	if (!interactive) {
```

Comment:
```suggestion
	if (json || isNonInteractiveOrCI()) {
```
to avoid the double negative :)

Reply:
5624941d